src/pybind/mgr/devicehealth/module.py

```diff
@@ -411,7 +411,7 @@ def put_device_metrics(self, ioctx, devid, data):
         self.log.debug('put_device_metrics device %s key %s = %s, erase %s' %
                        (devid, key, data, erase))
         with rados.WriteOpCtx() as op:
-            ioctx.set_omap(op, (key,), (str(json.dumps(data)),))
+            ioctx.set_omap(op, (key,), (json.dumps(data).encode('ascii'),))
```

Comment:
```suggestion
            ioctx.set_omap(op, (key,), (json.dumps(data).encode('utf-8'),))
```

As `utf-8` is the default in Python 3, I'd always use `utf-8` when decoding or encoding strings.

Reply:
OK,  agree with utf-8, thanks.

Reply:
i'am not sure whether you can directly apply your suggestion to this pr or i need to l manaully change locally, then push?

Reply:
Yeah, just amend your commit locally. If you press on [commit suggestion], github creates weird commits just containing my suggestion. I'd recommend against it.